File: RockitCompiler/benchmarks/bench_file_read.cpp
#include <cstdio>
#include <cstdint>
#include <fstream>
#include <string>

int main(int argc, char* argv[]) {
    std::ifstream f(argv[1]);
    std::string line;
    int64_t lines = 0;
    int64_t bytes = 0;
    while (std::getline(f, line)) {
        lines++;
        bytes += line.size() + 1; // +1 for newline
    }
    printf("%lld %lld\n", lines, bytes);
    return 0;
}

File: RockitCompiler/benchmarks/bench_file_read.rs
use std::io::{BufRead, BufReader};
use std::fs::File;
use std::env;

fn main() {
    let args: Vec<String> = env::args().collect();
    let f = File::open(&args[1]).unwrap();
    let reader = BufReader::new(f);
    let mut lines = 0i64;
    let mut bytes = 0i64;
    for line in reader.lines() {
        let l = line.unwrap();
        lines += 1;
        bytes += l.len() as i64 + 1; // +1 for newline
    }
    println!("{} {}", lines, bytes);
}
